src/averageAges.js

```diff
@@ -14,14 +13,20 @@
  *
  * @return {number}
  */
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
-}
+  const men = people.filter(person => {
+    const isMale = person.sex === 'm';
+
+    return century
+      ? isMale && Math.ceil(person.died / 100) === century
+      : isMale;
+  });
 
+  const calculateAge = men.map(person => person.died - person.born);
+
+  return calculateAverage(calculateAge);
```

Comment:
`calculateAverage` is a proper name for a function that do calculate it.
But `calculateAge` is not a function, but an array of ages instead.
It might seem minor, but when the codebase grows big, reasonable names make difference.

```suggestion
  const menAges = men.map(person => person.died - person.born);

  return calculateAverage(menAges);
```